torchvision/datasets/folder.py

```diff
@@ -38,15 +38,37 @@ def make_dataset(
     extensions: Optional[Tuple[str, ...]] = None,
     is_valid_file: Optional[Callable[[str], bool]] = None,
 ) -> List[Tuple[str, int]]:
+    """Generates a list of samples of a form (path_to_sample, class).
+
+    Args:
+        directory (str): root dataset directory
+        class_to_idx (Dict[str, int]): dictionary mapping class name to class index
+        extensions (Optional[Tuple[str, ...]], optional): A list of allowed extensions.
+            Both extensions and is_valid_file should not be passed. Defaults to None.
+        is_valid_file (Optional[Callable[[str], bool]], optional): A function that takes path of a file
+            and check if the file is a valid file
+            (used to check of corrupt files) both extensions and
+            is_valid_file should not be passed. Defaults to None.
+
+    Raises:
+        ValueError: In case ``extensions`` and ``is_valid_file`` is None.
```

Comment:
Technically, this is raised if `both_none` or `both_something`.

Reply:
addressed in a following commit